components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandler.java

```diff
@@ -340,6 +340,10 @@ protected void handlePostAuthentication(HttpServletRequest request,
                             }
                         }
                         authenticatedUserAttributes = FrameworkUtils.buildClaimMappings(mappedAttrs);
+                        if (isSPStandardClaimDialect(context.getRequestType(), appConfig) &&
```

Comment:
@IsuraD @mefarazath this will change the existing behavior of how the claims are released when requested claims are not there. Hope this is fine. Else we can do this through a config

Reply:
it is a bug to return claims without consent. So we don't need a config IMO.